### PR TITLE
Background job deleting indices

### PR DESCRIPTION
**Pull request by kafkas**

### CODE REVIEW

File: src/services/cleanup/CleanupService.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    constructor(private readonly app: FdrApplication) {}
+
+    public async deleteOldIndices(params: DeleteOldIndicesParams) {
+        const { limit, olderThanDays = 2 } = params;
+        const records = await this.db.prisma.overwrittenAlgoliaIndex.findMany({
+            take: limit,
+            where: {
+                overwrittenTime: {
+                    lte: new Date(Date.now() - olderThanDays * 24 * 60 * 60 * 1_000),
```
Comment (dsinghvi) [inline src/services/cleanup/CleanupService.ts:29]: nit: lets have a constant for `24 * 60 * 60 * 1_000` makes it easier to read (e.g. `MILLISECONDS_IN_ONE_DAY`)

File: src/services/cleanup/CleanupService.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+interface DeleteOldIndicesParams {
+    olderThanDays?: number;
+    limit?: number;
+}
+
+export interface CleanupService {
+    deleteOldIndices(params: DeleteOldIndicesParams): Promise<number>;
+}
+
+export class CleanupServiceImpl implements CleanupService {
```
Comment (dsinghvi) [inline src/services/cleanup/CleanupService.ts:12]: nit:  i could see us renaming it to be very specific (i.e. `AlgoliaIndexDeleterService`)

File: src/background.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import * as cron from "node-cron";
+import { type FdrApplication } from "./app";
+
+export function registerBackgroundTasks(app: FdrApplication) {
```
Comment (dsinghvi) [inline src/background.ts:4]: nit: lets rename this `registerAlgoliaIndexDeletionBackgroundTask`

**Comment by dsinghvi** [comment]
bunch of nits around naming -- in general, better to be specific than general and make it as readable for someone who isn't already in the codebase

File: src/controllers/docs/getDocsWriteV2Service.ts [hunk 4]
```diff
@@ -193,13 +193,15 @@ export function getDocsWriteV2Service(app: FdrApplication): WriteService {
...
                 if (prevDocs?.algoliaIndex) {
-                    await app.services.algolia.scheduleIndexDeletion(prevDocs.algoliaIndex);
+                    await app.services.db.prisma.overwrittenAlgoliaIndex.create({
+                        data: { indexId: prevDocs.algoliaIndex },
+                    });
                 }
             };
 
-            await deletePreviousIndex();
+            await schedulePreviousIndexDeletion();
```
[the remark this answers is not in the document]
Comment (kafkas) [reply src/controllers/docs/getDocsWriteV2Service.ts:204]: Yep, that would be better. A lot of other things can probably go into that.

[the remark this answers is not in the document]
Comment (dsinghvi) [reply src/controllers/docs/getDocsWriteV2Service.ts:204]: yup i think we'll slowly start putting all db related things in the db service

File: src/services/cleanup/CleanupService.ts [hunk 5]
```diff
@@ -1,3 +1,4 @@
+import { MILLISECONDS_IN_ONE_DAY } from "src/util";
```
Comment (dsinghvi) [inline src/services/cleanup/CleanupService.ts]: i just moved the const into this service (misc.ts is a bit weird)

**Pull request merged**